src/utils/document_extraction.py

```
@@ -0,0 +1,108 @@
+                return DocumentExtractionResult(
+                    text="\n\n".join(ocr_chunks),
+                    method="pdf_ocr",
+                    fallback_reason="PDF did not contain readable embedded text",
+                )
+
+            return DocumentExtractionResult(
+                text="",
+                method="pdf_no_text",
+                fallback_reason="No readable text found with embedded text extraction or OCR",
+            )
+    except Exception as exc:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)